src/expr.rs

```diff
@@ -116,10 +122,37 @@ impl Rewrite for ast::Expr {
                 };
                 Some(format!("continue{}", id_str))
             }
+            ast::Expr_::ExprBreak(ref opt_ident) => {
+                let id_str = match *opt_ident {
+                    Some(ident) => format!(" {}", ident),
+                    None => String::new(),
+                };
+                Some(format!("break{}", id_str))
+            }
             ast::Expr_::ExprClosure(capture, ref fn_decl, ref body) => {
                 rewrite_closure(capture, fn_decl, body, self.span, context, width, offset)
             }
-            _ => context.codemap.span_to_snippet(self.span).ok(),
+            _ => {
+                // We do not format these expressions yet, but they should still
+                // abide by our width restrictions.
+                let snippet = context.codemap.span_to_snippet(self.span).unwrap();
+
+                {
+                    let mut lines = snippet.lines();
+
+                    if lines.next().unwrap().len() > width {
+                        return None;
+                    }
```

Comment:
I think the width constaint is only for the last line, we don't have (yet) an explicit width constraint for other lines (we currently use `config.max_width` for that).